azavea/pwd-fishway
# Add meet the fish modal

## Overview

Adds a modal component for "Meet the Fish"

### Demo

![localhost_3474_ (1)](https://user-images.githubusercontent.com/4432106/57721954-6cb9f480-7653-11e9-9491-1beaa9713e4c.png)
![localhost_3474_ (2)](https://user-images.githubusercontent.com/4432106/57721955-6cb9f480-7653-11e9-9ede-010d21b94e9b.png)
![localhost_3474_ (3)](https://user-images.githubusercontent.com/4432106/57721957-6d528b00-7653-11e9-96b3-a9d9ca7209e3.png)


### Notes

 - There is a temporary "Meet a fish" button shown on the page. Once we complete this page the intent is that each fish image will act as a "button" to open the modal, setting the index of the fish it corresponds to in its props.
 - Styling is somewhat there but is still _very_ rough and dependent on the screen being large enough for the content.
 - We're missing text content for American Eel habitats (it wasn't in the original exhibit either). I checked with @alexelash and her preference was to hide the link/section entirely for eels.
 - I looked into using the `react-scroll` library for the animated scrolling, but decided to instead rely on the native browser scrolling to keep things simpler. We could look into switching to this if we want fancier scrolling animations.

## Testing Instructions

 * `./scripts/update`
 * `./scripts/server`
 * Clicking "Meet a fish" button on the Meet the Fish tab should open the modal.
 * The links in the left sidebar should animate the content area to show that section
 * The navigation buttons in the bottom should move between fish. Moving to view a different fish should reset the scrollable content to the top
 * Opening and closing the Meet the Fish modal should reset it back to the starting fish.
 * Changing the `index` prop given to `MeetTheFishModal` in `MeetTheFish.js` should cause the modal to open to a different starting fish.
